### PR TITLE
Fix `Studio.queueCallback` after interpreter upgrade

### PR DESCRIPTION
**Pull request by joshlory**
Thank you @balderdash for catching this issue!

This applies the same fix as https://github.com/code-dot-org/code-dot-org/pull/15485/commits/ad78b86756c28a956b2c02b62f7338f0dfbef349#diff-11d38f100f6df9c907f72e0cf6bb3671R416.

### CODE REVIEW

File: apps/src/studio/studio.js [hunk 1]
```diff
@@ -5353,6 +5353,11 @@ Studio.queueCallback = function (callback, args) {
   };
 
   registerEventHandler(Studio.eventHandlers, handlerName, () => {
+    // remove the last argument because stepCallExpression always wants to push it back on.
+    if (state.arguments.length > 0) {
+      state.value = state.arguments.pop();
+    }
+
     const depth = Studio.interpreter.stateStack.unshift(state);
```
Comment (pcardune) [inline apps/src/studio/studio.js:5361]: Dang. I thought JSInterpreter was the only thing that monkeyed with interpreter internals. This next line is going to be broken too with the PR that I have out right now but which is not merged yet. I guess I will try to write tests for this and/or factor this stuff out of `studio.js`

Comment (joshlory) [reply apps/src/studio/studio.js:5361]: Yeah this code is super gross 😕.  I was hoping we could replace the immediate callback with `await` or `Promise` queues but it's not on the horizon until after back-to-school.

**Comment by joshlory** [comment]
This PR is on hold since PR https://github.com/code-dot-org/code-dot-org/pull/15485 has been reverted.

**Comment by pcardune** [comment]
This is now superseded by #15702

**Pull request closed without merging**